clients/drcachesim/tools/basic_counts.h

```diff
@@ -143,6 +147,7 @@ class basic_counts_t : public analysis_tool_t {
         memref_tid_t tid = 0;
         // A vector to support windows.
         std::vector<counters_t> counters;
```

Comment:
So we already have support for a separate set of counters per trace window: it seems natural to do the same for a time series, with a vector of statistics printed out at the end -- or to even use the same vector (abandoning per-window if time series is requested, and tweaking the output to say "Snapshot #" instead of "Window #".

Reply:
On this topic: with the current quantum by absolute timestamp, we won't have regular heartbeats: how should a tool handle that?  Record the index with each snapshot?

Reply:
This also points at a possible alternative to a new tool API: inserting a synthetic marker record like WINDOW_ID today: SNAPSHOT_ID or something.  Not sure it's better but worth thinking about.

Reply:
> and tweaking the output to say "Snapshot #" instead of "Window #".

Done. Printing both the snapshot deltas and snapshots.

> with the current quantum by absolute timestamp, we won't have regular heartbeat

If the tool cares to note the lull points in the trace, it could observe the gaps in the quantum_ids passed to it. The new basic counts example shows that.